source/agora/test/Base.d

```diff
@@ -776,29 +775,28 @@ public class TestAPIManager
     ***************************************************************************/
 
     public void waitForPreimages (const(Enrollment)[] enrolls, Height height,
-        Duration timeout = 10.seconds)
+        Duration timeout = 10.seconds, string file = __FILE__, int line = __LINE__)
     {
-        this.waitForPreimages(iota(this.test_conf.node.test_validators), enrolls, height, timeout);
+        this.waitForPreimages(iota(this.test_conf.node.test_validators), enrolls, height, timeout, file, line);
     }
 
     /// Ditto
     public void waitForPreimages (Idxs)(Idxs clients_idxs,
         const(Enrollment)[] enrolls, Height height,
-        Duration timeout = 10.seconds)
+        Duration timeout = 10.seconds, string file = __FILE__, int line = __LINE__)
     {
         static assert (isInputRange!Idxs);
         import std.algorithm.searching : any;
 
         foreach (idx; clients_idxs)
-            foreach (idx_enroll, enroll; enrolls)
-                if (clients_idxs.canFind(idx_enroll))
-                {
-                    scope node = this.nodes[idx];
-                    retryFor(node.client.getPreimages(Set!Hash.from(enroll.utxo_key.only))
-                        .any!(preimage => preimage.height >= height),
-                            timeout, format!"Client #%s (%s) has no preimage at height %s for node enrolled with utxo %s"
-                                (idx, node.address, height, enroll.utxo_key));
-                }
+        {
+            scope node = this.nodes[idx];
+            foreach (enroll; enrolls)
+                retryFor(node.client.getPreimages(Set!Hash.from(enroll.utxo_key.only))
+                    .any!(preimage => preimage.height >= height),
+                        timeout, format!"Client #%s (%s) has no preimage at height %s for node enrolled with utxo %s"
```

Comment:
```suggestion
                        timeout, format!"Client #%s (%s) has no preimage for height %s for node enrolled with utxo %s"
```
I suggest it because it looks like the current height. But you could determine better than me for that.

Reply:
I agree will change.

Reply:
Updated